### PR TITLE
E.164 phone number generator

### PR DESCRIPTION
**Pull request by daleattree**
Added E.164 support as requested in issue #668

### CODE REVIEW

File: test/Faker/Provider/ProviderOverrideTest.php [hunk 1]
```diff
@@ -112,6 +112,7 @@ public function testPhoneNumber($locale = null)
         $faker = Faker\Factory::create($locale);
 
         $this->assertRegExp(static::TEST_STRING_REGEX, $faker->phoneNumber);
+        $this->assertRegExp(static::TEST_STRING_REGEX, $faker->e164PhoneNumber);
```
Comment (fzaninotto) [inline test/Faker/Provider/ProviderOverrideTest.php]: Please put this test in a new method

Comment (fzaninotto) [inline test/Faker/Provider/ProviderOverrideTest.php]: Also, testing that it's a string is too loose. Please validate that the returned value complies with the expected format.

File: test/test.php [hunk 2]
```diff
@@ -10,6 +10,7 @@
 <?php for ($i=0; $i < 10; $i++): ?>
   <contact firstName="<?php echo $faker->firstName ?>" lastName="<?php echo $faker->lastName ?>" email="<?php echo $faker->email ?>" >
     <phone number="<?php echo $faker->phoneNumber ?>"/>
+    <phone number="<?php echo $faker->e164PhoneNumber ?>" format="E164"/>
```
Comment (fzaninotto) [inline test/test.php]: please remove this line

**Comment by fzaninotto** [comment]
You must also update the project's README to mention the new formatter.

**Comment by daleattree** [comment]
Refactored as requested.

File: readme.md [hunk 3]
```diff
@@ -127,6 +127,11 @@ Each of the generator properties (like `name`, `address`, and `lorem`) are calle
     paragraphs($nb = 3)     // array('Quidem ut sunt et quidem est accusamus aut. Fuga est placeat rerum ut. Enim ex eveniet facere sunt.', 'Aut nam et eum architecto fugit repellendus illo. Qui ex esse veritatis.', 'Possimus omnis aut incidunt sunt. Asperiores incidunt iure sequi cum culpa rem. Rerum exercitationem est rem.')
     text($maxNbChars = 200) // 'Fuga totam reiciendis qui architecto fugiat nemo. Consequatur recusandae qui cupiditate eos quod.'
 
+### `Faker\Provider\PhoneNumber`
```
Comment (fzaninotto) [inline readme.md]: It's already documented in `Faker\Provider\en_US\PhoneNumber`, document the new method there

File: test/Faker/Provider/PhoneNumberTest.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace Faker\Test\Provider;
+
+use Faker\Generator;
+use Faker\Provider\PhoneNumber;
+
+class PhoneNumberTest extends \PHPUnit_Framework_TestCase
+{
+
+    const TEST_E164_REGEX = '/^\+[0-9]{11}$/';
```
Comment (fzaninotto) [inline test/Faker/Provider/PhoneNumberTest.php]: not used

Comment (daleattree) [inline test/Faker/Provider/PhoneNumberTest.php]: Done.

**Comment by fzaninotto** [comment]
Can you please squash your commits?

**Comment by daleattree** [comment]
I tried. Not sure if I succeed.

I did a rebase but not sure if I achieved the desired result.

If it is still not what you are looking for, could you please point me in
the right direction?

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**